src/OpenLoco/Windows/Vehicle.cpp

```diff
@@ -1,177 +1,3547 @@
+                    }
+
+                    // TODO: define constant for hot zone
+                    if (pos.y < vehicleWindow->widgets[widx::carList].top + vehicleWindow->y + 5)
+                    {
+                        Ui::ScrollView::verticalNudgeUp(vehicleWindow, vehicleWindow->getScrollDataIndex(widx::carList), widx::carList);
+                    }
+                    else if (pos.y > vehicleWindow->widgets[widx::carList].bottom + vehicleWindow->y - 5)
+                    {
+                        Ui::ScrollView::verticalNudgeDown(vehicleWindow, vehicleWindow->getScrollDataIndex(widx::carList), widx::carList);
+                    }
+                    break;
+                }
+            }
+            if (vehicleWindow->isActivated(widx::remove))
+            {
+                vehicleWindow->activated_widgets &= ~(1ULL << widx::remove);
+                WindowManager::invalidateWidget(WindowType::vehicle, vehicleWindow->number, widx::remove);
+            }
+        }
+
+        void scrollDragEnd(const Gfx::point_t& pos)
+        {
+            auto vehicleWindow = getVehicleDetailsWindow(pos);
+            if (vehicleWindow == nullptr && _dragCarComponent != nullptr)
+            {
+                return;
+            }
+
+            auto targetWidget = vehicleWindow->findWidgetAt(pos.x, pos.y);
+            switch (targetWidget)
+            {
+                case widx::remove:
+                    GameCommands::do_6((*_dragCarComponent)->id);
+                    break;
+                case widx::carList:
+                {
+                    auto car = getCarFromScrollViewPos(*vehicleWindow, pos);
+                    if (car != nullptr)
+                    {
+                        GameCommands::do_0((*_dragCarComponent)->id, car->id);
+                    }
+                    break;
+                }
+            }
+        }
+    }
+
+    namespace Cargo
+    {
+        static void onRefitButton(window* const self, const widget_index wi);
+
+        static bool canRefit(OpenLoco::vehicle_head* headVehicle)
+        {
+            if (!isPlayerCompany(headVehicle->owner))
+            {
+                return false;
+            }
+
+            OpenLoco::Things::Vehicle::Vehicle train(headVehicle);
+
+            if (train.cars.empty())
+            {
+                return false;
+            }
+
+            auto object = ObjectManager::get<vehicle_object>(train.cars.firstCar.front->object_id);
+            return (object->flags & FlagsE0::refittable);
+        }
+
+        // 004B3DDE
+        static void prepareDraw(window* const self)
+        {
+            if (self->widgets != widgets)
+            {
+                self->widgets = widgets;
+                self->initScrollWidgets();
+            }
+
+            Common::setActiveTabs(self);
+
+            auto* headVehicle = Common::getVehicle(self);
+            auto ax = headVehicle->var_44;
```

Comment:
You could probably skip these intermediate variables

Reply:
I'm going to go through and change all of these in a latter pr